### PR TITLE
Feature/route setter

### PR DESCRIPTION
**Pull request by Robert-W**
Feature(s):
* Routes are now autogenerated based on some config settings.
* Metadata is always included.
* Resource routes are only added if there is a valid configuration passed in. Otherwise, they are ignored.

Breaking Changes:
* OAuth will not be working locally until a follow up PR is added.
* Each route needs config options for the route setter to know how to set it up and find the configuration.
* Controllers now receive an options argument instead of multiple arguments. Since the controllers may not all accept the same args in the same order, making it an object simplifies things.

Remaining Items to be done in separate PR:
* Move oauth configuration to new location. It is currently passed in via a dstu2 configuration and oauth should not be spec specific. We also need to update documentation to match this. This should also be setup in the express lib since the logic to include this and the middleware it uses is different from all the other routes. OAuth may also be moved to another repo and pulled in as a npm dependency for testing, but is not really part of core. More thought needs to go into this.

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -34,6 +34,11 @@ exports.STU3 = {
 	}
 };
 
+exports.CONFIG_KEYS = {
+	OBSERVATION: 'observation',
```
Comment (andrewmarcus) [inline src/constants.js]: Do we need to add keys here for every resource we create?

Comment (Robert-W) [reply src/constants.js]: Yes, we need a mapping from the user provided config to a resource. If you look at node-fhir-server-mongo or the runner file in this repo. You can see an example of where these are used. E.g.

```javascript
injected_config = {
  profiles: {
    dstu2: {
      patient: { ...patient config },
      observation: { ...observation config }
    }
  }
}
```

**Comment by Robert-W** [comment]
We need the constants for the conformance statements for now. Each endpoint needs to be able to describe itself for the metadata endpoint (Capability statement) to work. Where the constants live now is a temporary solution (for the resource specific ones) until all the resources are actually generated because when we have valid resources, the endpoints can get any info from the resources.  Some resources in our app hosted on aws are not auto generated and are manually built using these.

**Comment by Robert-W** [comment]
Closing this in favor of a newer PR

**Pull request closed without merging**